coderd/users.go

```diff
@@ -1056,6 +1068,9 @@ func (api *API) postLogin(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// key := httpmw.APIKey(r)
+	aReq.New = *key
+
```

Comment:
Instead of returning the API key from `createAPIKey`, I would like to be able to call `key := httpmw.APIKey(r)`.
If I had the ability to call this function, I would also be able to [get the user in the audit request](https://github.com/coder/coder/pull/5925/files#diff-cc84da01e587fa9b8971d66c92f68321bad61d76a161a36926ccfe6d57277048R165) (right now, the user is always `undefined`). However, calling this function gives me the following error: `developer error: ExtractAPIKey middleware not provided`.

Reply:
Yea, you cannot use `httpmw.APIKey` in `postLogin` because the `postLogin` is making the apikey. Using `ExtractAPIKey` in the MW assumes the token is set on the request via a cookie or header. Chicken and the egg problem :chicken: :egg: 

Where does the audit user id get set usually? I cannot find the code that sets the audit log UserID. I only see: https://github.com/coder/coder/blob/1acc73c016a0e68d61231ba5bfcdb2e5fd218b4a/coderd/audit/request.go#L165-L165

If I knew where userid was set, I could help more :thinking:

Reply:
I just saw the other comment, I see where it is set.